client/src/main/java/io/pravega/client/stream/impl/ReaderGroupImpl.java

```diff
@@ -185,10 +192,25 @@ private Checkpoint completeCheckpoint(String checkpointName) {
 
     @Override
     public void resetReaderGroup(ReaderGroupConfig config) {
+        manageSubscriptions(config);
         Map<SegmentWithRange, Long> segments = getSegmentsForStreams(controller, config);
         synchronizer.updateStateUnconditionally(new ReaderGroupStateInit(config, segments, getEndSegmentsForStreams(config)));
     }
 
+    private void manageSubscriptions(ReaderGroupConfig config) {
+        ReaderGroupConfig oldConfig = getReaderGroupConfig();
+        Set<Stream> newStreams = config.isSubscriberForRetention() ? config.getStartingStreamCuts().keySet() : Collections.emptySet();
+        Set<Stream> oldStreams = oldConfig.isSubscriberForRetention() ? oldConfig.getStartingStreamCuts().keySet() : Collections.emptySet();
+
+        Set<Stream> streamsToSub = Sets.difference(newStreams, oldStreams);
+        Set<Stream> streamsToUnsub = Sets.difference(oldStreams, newStreams);
+
+        // Unsubscribe to older streams
+        streamsToUnsub.forEach(s -> getAndHandleExceptions(controller.deleteSubscriber(scope, s.getStreamName(), groupName), RuntimeException::new));
+        // Subscribe to newer streams
+        streamsToSub.forEach(s -> getAndHandleExceptions(controller.addSubscriber(scope, s.getStreamName(), groupName), RuntimeException::new));
```

Comment:
Please use `Futures.getThrowingException()`. This would avoid wrapping the exception within RuntimeException.

The same comment applies to other methods too.

Reply:
Used `getThrowingException` instead.